File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
# DFS
def DFS(V):
    visited_DFS[V] = 1 #방문처리
    DFS_result.append(V)
    for node in graph[V]:
        if not visited_DFS[node]:
            DFS(node)


# BFS
def BFS(V):
    queue = [V]
    visited_BFS[V] = 1 #방문처리
    while queue:
        V = queue.pop(0)
        BFS_result.append(V)
        for node in graph[V]:
            if not visited_BFS[node]:
                visited_BFS[node] = 1 # 방문처리
                queue.append(node)


N, M, V = map(int, input().split())

# 인접 리스트
graph = [[] for _ in range(N+1)]
for i in range(M):
    s, e = map(int, input().split())
    graph[s].append(e)
    graph[e].append(s)

for row in range(N+1):
    graph[row].sort()

# 방문
visited_DFS = [0]*(N+1)
visited_BFS = [0]*(N+1)

DFS_result = []
BFS_result = []

DFS(V)
BFS(V)

print(*DFS_result)
print(*BFS_result)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.awt.print.Pageable;
import java.util.*;

public class Main {

    static boolean visited[];
    static ArrayList<Integer>[] arr;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int start = sc.nextInt();

        arr = new ArrayList[n+1];
        for (int i = 1; i <= n; i++){
            arr[i] = new ArrayList<Integer>();
        }

        for (int i = 0; i < m; i++) {
            int s = sc.nextInt();
            int e = sc.nextInt();
            arr[s].add(e);
            arr[e].add(s);
        }

        // 번호가 적은것을 먼저 방문하기 위해 정렬하기
        for (int i = 1; i <= n; i++){
            Collections.sort(arr[i]);
        }
        visited = new boolean[n+1];
        dfs(start);
        System.out.println();
        visited = new boolean[n+1];
        bfs(start);
        System.out.println();
    }

    public static void dfs(int node){
        System.out.print(node + " ");
        visited[node] = true;
        for (int i : arr[node]){
            if (!visited[i]){
                dfs(i);
            }
        }
    }

    public static void bfs(int node){
        Queue<Integer> queue = new LinkedList<Integer>();
        queue.add(node);
        visited[node] = true;

        while(!queue.isEmpty()) {
            int now_node = queue.poll();
            System.out.print(now_node + " ");
            for (int i : arr[now_node]){
                if (!visited[i]){
                    visited[i] = true;
                    queue.add(i);
                }
            };
        }
    }
}
